cognite_toolkit/_cdf_tk/warnings.py

```diff
@@ -52,3 +64,40 @@ def __str__(self) -> str:
             for warning in group_list:
                 output.append(f"{'    ' * 2} * {warning!s}")
         return "\n".join(output)
+
+
+@dataclass(frozen=True)
+class GeneralWarning(ToolkitWarning):
+    severity: SeverityLevel | None = None
+    message: str | None = None
+    details: Union[None, str, list[str]] = None  # Allow None, str, list[str]
+
+    def __str__(self) -> str:
+        output = [f"{RICH_WARNING_FORMAT}{self.severity}{type(self).__name__}: {self.message}"]
+
+        if self.details:
+            if isinstance(self.details, str):
+                output.append(f"{'    ' * 2}{self.details}")
+            else:
+                for detail in self.details:
+                    output.append(f"{'    ' * 2}{detail}")
+        return "\n".join(output)
+
+
+@dataclass(frozen=True)
+class ToolkitDependenciesIncludedWarning(GeneralWarning):
+    severity: SeverityLevel = SeverityLevel.LOW
+    message: str = "Some resources were added due to dependencies."
```

Comment:
Not sure if this is possible, but in this case message is a property of the warning class and not intended to be changed when you use the warning.
```suggestion
    message: ClassVar[str] = "Some resources were added due to dependencies."
```

Reply:
In addition, should probably include which dependencies are included?